mathics/builtin/system.py

```diff
@@ -407,7 +407,11 @@ class UserName(Predefined):
     name = "$UserName"
 
     def evaluate(self, evaluation) -> String:
-        return String(os.getlogin())
+        try:
+            return String(os.getlogin())
+        except:
+            import pwd
+            return String(pwd.getpwuid(os.getuid())[0])
```

Comment:
@mmatera his should be a separate PR and merged in.

Reply:
See PR  #1163.  If pass CI, let's merge it.